plotnine/tests/test_ggplot_internals.py

```diff
@@ -77,6 +77,17 @@ def test_ggplot_parameters():
         ggplot([1, 2, 3], aes('x'))
 
 
+def test_ggplot_parameters_grouped():
+    p = ggplot(df.groupby('x'), aes('x'))
+
+    assert p.data is df
+    assert p.mapping == aes('x')
+    
+    p = ggplot(data=df, mapping=aes('x'))
+    assert p.data is df
+    assert p.mapping == aes('x')
```

Comment:
I am wondering why not `p.mapping == aes('x', group='x')`. Would it be overly presumptuous to turn the pandas groups into a mapping to the group variable?

Reply:
Ah, good call--it looks like this is what ggplot does...

https://github.com/tidyverse/ggplot2/blob/d700a0b99ad9776bd9fcc17ff568c182c458c3ee/R/fortify.r#L37

Reply:
I don't know what ggplot does, but I had this functionality
https://github.com/has2k1/plotnine/blob/2b0ccdba4ed385eacaedfbb1d410ec55d10bbbfa/plotnine/ggplot.py#L60-L63
for plydata. 

However, there is an inconsistency which I realised because of this PR. It only works when passed through the `ggplot()` call and I do not think it is worth it to add more code. So it may be dropped entirely.